syncto/authentication.py

```diff
@@ -45,15 +48,19 @@ def build_sync_client(request):
     cache_key = 'credentials_%s' % utils.hmac_digest(hmac_secret,
                                                      bid_assertion)
 
-    credentials = cache.get(cache_key)
+    encrypted_credentials = cache.get(cache_key)
 
-    if not credentials:
+    if not encrypted_credentials:
         ttl = int(settings['syncto.cache_credentials_ttl_seconds'])
         tokenserver = TokenserverClient(bid_assertion, client_state)
         if statsd:
             statsd.watch_execution_time(tokenserver, prefix="tokenserver")
         credentials = tokenserver.get_hawk_credentials(duration=ttl)
-        cache.set(cache_key, credentials, ttl)
+        encrypted = encrypt(json.dumps(credentials), client_state, hmac_secret)
+        cache.set(cache_key, encrypted, ttl)
```

Comment:
I have added the two spec tests here: https://github.com/mozilla-services/syncto/pull/30/files#diff-54b1188b370166a8a9ba337f3dce6711R66